src/main/java/hudson/security/GlobalMatrixAuthorizationStrategy.java

```diff
@@ -273,6 +275,9 @@ public AuthorizationStrategy newInstance(StaplerRequest req, JSONObject formData
                     }
                     if ((Boolean) e.getValue()) {
                         Permission p = Permission.fromId(e.getKey());
+                        if (p == null) {
+                            throw new IllegalArgumentException("Unknown permission:\"" + e.getKey() + "\" set to \"" + e.getValue() +"\" for sid \"" + sid + "\"");
+                        }
```

Comment:
This is wrong. It should simply skip over the nonexistent permission; it is a normal condition, for example when you disable a plugin you had previously been using. At most the condition should be logged at `FINE`.